fujiji-client/src/components/ListingForm/ListingForm.jsx

```diff
@@ -206,7 +206,24 @@ export default function ListingForm({
       setUploadErrorMessage('Allowed files: .png, .jpg, .jpeg');
     } else {
       setUploadedImage(uploadedFile);
-      setListingImageUrl(URL.createObjectURL(uploadedFile));
+      const url = await onImageUpload(uploadedFile);
+      if (url.error) {
+        toast({
+          title: 'Oops! Unable to upload the picture, We are working on it!',
+          status: 'error',
+          duration: 9000,
+          isClosable: true,
+        });
+      } else {
+        url.then((result) => setListingImageUrl(result.data.imageUrl)).catch(() => {
```

Comment:
We also need to change this now, since the url is not a promise anymore. 

Please refer to comment above

Reply:
yeah yeah my bad